Optional maximum wave attenuation

daleeidd opened the pull request:
One solution for #726.

Adds a maximum depth value for wave attenuation. Anything below this value will have no attenuation. Drawbacks is that wave attenuation is overall less effective. Moderate spectrums seem to work well enough. Completely optional as if the value is set to the maximum, the interpolation is skipped so this is a safe change.

TODO: documentation and release notes

huwb commented:
I tried it and i think it works really well. i stress tested it and i had to go fairly extreme to get waves intersecting with the underlying terrain, is this what you refer to by "overall less effective"? i think it might be good to specify what hte drawback can be in the tooltip as its not necessarily clear what less effective means in practice.

i've been thinking about how to expose the parameter. its under the "Attenuation in Shallows". i wonder if maybe calling it "Shallow Max Depth" or something, and perhaps tweaking the tooltip to say "Any water deeper than this will receive full wave strength". i'm not sure these suggestions are perfect but just throwing them in the mix.

huwb commented:
I'm a thumbs up for merging once you're happy but i am happy to re-review later if it helps as well

huwb commented on crest/Assets/Crest/Crest/Scripts/LodData/LodDataMgrSeaFloorDepth.cs:
```diff
@@ -16,11 +16,13 @@ namespace Crest
     /// </summary>
     public class LodDataMgrSeaFloorDepth : LodDataMgr
     {
+        internal const float k_DepthBaseline = 1_000f;
```
may be worth also referencing the constant in the shader code from here (you have already referenced in the other direction)

daleeidd replied:
Done. Thanks.

huwb commented on crest/Assets/Crest/Crest/Scripts/LodData/Settings/SimSettingsAnimatedWaves.cs:
```diff
@@ -27,6 +27,11 @@ public partial class SimSettingsAnimatedWaves : SimSettingsBase
         public float AttenuationInShallows => _attenuationInShallows;
 
+        [Tooltip("Any deeper than this value will have no attenuation (ie full wave spectrum). Set to maximum value (1,000) to disable. If it is used, it will reduce the overall attenuation effectiveness.")]
+        [SerializeField, Range(0f, LodDataMgrSeaFloorDepth.k_DepthBaseline)]
```
it may be we should not allow this to be 0 to avoid div by 0, (or clamp it before giving it to the shader)

daleeidd replied:
I made minimum 1. I doubt anyone wants anything below that and nice and round.

daleeidd commented:
> I tried it and i think it works really well. i stress tested it and i had to go fairly extreme to get waves intersecting with the underlying terrain, is this what you refer to by "overall less effective"? i think it might be good to specify what hte drawback can be in the tooltip as its not necessarily clear what less effective means in practice.

Thanks. Pretty much. I've updated it. LMKWYT.

Also addressed other feedback.

huwb reviewed (approved):
Looks good

huwb commented:
Nice one

daleeidd commented:
Thanks.
